src/main/java/com/treasuredata/client/TDRequestErrorHandler.java

```diff
@@ -194,6 +195,12 @@ else if (e instanceof SSLException) {
                 return new TDClientSSLException(sslException);
             }
         }
+        else if (e instanceof StreamResetException) {
+            // okhttp 4.10.0 will throw 429 Too Many Requests and defaultHttpResponseErrorResolver will handle
+            // just retry after 1 secs but we could consider increasing it
+            Date retryAfter = new Date(System.currentTimeMillis() + TimeUnit.SECONDS.toMillis(1));
+            return clientError(new TDClientHttpTooManyRequestsException(e.getMessage(), retryAfter));
```

Comment:
Just throw TDClientHttpTooManyRequestsException without wrapping it with clientError. clientError is just used for suppressing warning messages. In this case, we just can use the default retry backoff without specifying retryAfter parameter.

Reply:
I see, if we can throw `TDClientHttpTooManyRequestsException` , we don't have to modify `clientError` as `responseContext` is not available at this point